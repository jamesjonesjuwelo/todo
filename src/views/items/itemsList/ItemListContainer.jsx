import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import styled from 'styled-components';
import { createNewItem, getList } from '../../../store/action/notesAction';
import bindActionCreators from '../../../../util/bindActionCreators';
import { ItemComponentRoute } from '../item/ItemComponent';

export default class ItemListContainer extends Component {

  componentDidMount() {
    this.props.getList();
  }

  render() {
    const { list } = this.props;

    return (
      <Container>
        {list.length
          ? list.map(item => <ItemComponentRoute key={`item-${item.id}`} item={item} />)
          : <NoItems>There are no list items.</NoItems>
        }
      </Container>
    )
  }
}

const mapStateToProps = (state) => ({
  list: state.notesReducer.list
});

const mapDispatchToProps = (dispatch, props) => bindActionCreators({
  createNewItem,
  getList
}, dispatch, props);

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps)
);

const transition = `
    -moz-transition: all 0.15s ease-in;
    -o-transition: all 0.15s ease-in;
    -webkit-transition: all 0.15s ease-in;
    transition: all 0.15s ease-in;
`;

const Container = styled.div`
  height: 100%;
  width: calc(100% - 70px);
  margin-left: 70px;
  overflow: scroll;
`;

const NoItems = styled.p`
  text-align: center;
`;


export const ItemListContainerRoute = enhance(ItemListContainer);