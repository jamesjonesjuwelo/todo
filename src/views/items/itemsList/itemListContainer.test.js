import React from 'react';
import { shallow } from 'enzyme';
import ItemListContainer from './ItemListContainer';
import 'jest-styled-components';

const props = {
  list: [
    {
      id: 1,
      name: 'Test Name',
      description: 'Test Description',
      createdAt: 'Test Date',
      completed: false
    },
    {
      id: 2,
      name: 'Test2 Name',
      description: 'Test2 Description',
      createdAt: 'Test2 Date',
      completed: false
    }
  ],
  getList: jest.fn()

};

describe('Item List Container', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <ItemListContainer {...props} /> );
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('container children length', () => {
    expect(wrapper.children().length).toBe(props.list.length);
  })
});