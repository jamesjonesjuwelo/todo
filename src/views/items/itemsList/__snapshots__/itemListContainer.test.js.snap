// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item List Container renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListContainer
    getList={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    list={
      Array [
        Object {
          "completed": false,
          "createdAt": "Test Date",
          "description": "Test Description",
          "id": 1,
          "name": "Test Name",
        },
        Object {
          "completed": false,
          "createdAt": "Test2 Date",
          "description": "Test2 Description",
          "id": 2,
          "name": "Test2 Name",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(ItemComponent)
          item={
            Object {
              "completed": false,
              "createdAt": "Test Date",
              "description": "Test Description",
              "id": 1,
              "name": "Test Name",
            }
          }
        />,
        <Connect(ItemComponent)
          item={
            Object {
              "completed": false,
              "createdAt": "Test2 Date",
              "description": "Test2 Description",
              "id": 2,
              "name": "Test2 Name",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "item-1",
        "nodeType": "class",
        "props": Object {
          "item": Object {
            "completed": false,
            "createdAt": "Test Date",
            "description": "Test Description",
            "id": 1,
            "name": "Test Name",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "item-2",
        "nodeType": "class",
        "props": Object {
          "item": Object {
            "completed": false,
            "createdAt": "Test2 Date",
            "description": "Test2 Description",
            "id": 2,
            "name": "Test2 Name",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(ItemComponent)
            item={
              Object {
                "completed": false,
                "createdAt": "Test Date",
                "description": "Test Description",
                "id": 1,
                "name": "Test Name",
              }
            }
          />,
          <Connect(ItemComponent)
            item={
              Object {
                "completed": false,
                "createdAt": "Test2 Date",
                "description": "Test2 Description",
                "id": 2,
                "name": "Test2 Name",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "item-1",
          "nodeType": "class",
          "props": Object {
            "item": Object {
              "completed": false,
              "createdAt": "Test Date",
              "description": "Test Description",
              "id": 1,
              "name": "Test Name",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "item-2",
          "nodeType": "class",
          "props": Object {
            "item": Object {
              "completed": false,
              "createdAt": "Test2 Date",
              "description": "Test2 Description",
              "id": 2,
              "name": "Test2 Name",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
